presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -191,7 +191,7 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
                     partitions,
                     compactEffectivePredicate,
                     effectivePredicate,
-                    TupleDomain.none(),
+                    TupleDomain.all(),
```

Comment:
Why is this change? Given `partitionColumns.isEmpty()`, this should be a `none()`?

Reply:
empty() mean it should be all --no filter. "none" here means empty result. My understanding is that.. none intersect filter = none, so if there are no partition col filter and there are other filters it should all intersect filter = filter

Reply:
This patch looks good to me except this change. Can we file another PR only with this change with some test coverage?

Reply:
@highker The following test "testGroupByKeyPredicatePushdown" fails without this change. By enabling this test with pushdown I found this bug and so fixed it as part of this PR.

Reply:
@arhimondr, @rschlussel, could you help verifying this codepath? It sounds to me the logic is correct. But I'm just surprise this has been there for years without an issue?

Reply:
I'm not sure if this is logically correct. 

What is set here is `enforcedConstraint`. In other words the variable stores predicates enforced by selecting specific partitions.

Assuming there's a table

```
CREATE TABLE t (
  c BIGINT
) WITH (
  partitioned_by = ARRAY['ds']
)
```

That has two partitions: `ds=2018-01-01`, and `ds=2018-01-01`

And there's a query:

```
SELECT *
FROM t
WHERE ds = '2018-01-01' and c = 123
```

The `HivePartitionManager` supposed to select only a single partition to scan (`ds=2018-01-01`) and set the `enforcedContraint` to `ds=2018-01-01` in order to make sure engine does not evaluate this. The `unenforcedConstraint` (`effectivePredicate`) should be set to `c = 123`, to make sure that the engine evaluates that.

Now let's consider the table is unpartitioned. With unpartitioned table there's no possible constrain that the `HivePartitionManager` can enforce. As there are no partitions to "select" or "skip". Thus the `enforcedContraint` should be set to `TupleDomain.none()`.

I don't know exactly what's the problem with the failing tests. But i would recommend digging deeper to it. As changing this may have other negative side effects.